src/main/java/org/opensearch/security/securityconf/impl/AllowlistingSettings.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright OpenSearch Contributors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package org.opensearch.security.securityconf.impl;
+
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestChannel;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestStatus;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class AllowlistingSettings {
```

Comment:
Changed

Reply:
`WhitelistSetting` can extends `AllowlistingSettings`, but none of the methods can be removed from `WhitelistSetting` even with extends. Only the getters and setters can be reused, but even they cannot be removed from the class because serialization and deserialization relies on getters and setters. Without getters and setters, serialization and deserilizaiton won't work.

Reply:
My concern is the method `checkRequestIsAllowed(...)` as it contains logic that is material to the access control systems, if you can't use inheritance, could we use composition to reuse this logic between the systems?

Reply:
Can you elaborate your concern? I don't see where the concern is. `WhitelistingSettings` and `AllowlistingSettings` are 2 different things without any dependencies with each other in this case. They are evaluated separately in permission validation. I'm not saying that I cannot make `WhitelistingSettings` extends `AllowlistingSettings`, but I don't see the value to do that here.

Reply:
I made `WhitelistingSettings` extend `AllowlistingSettings` if the resolves your concern.

Reply:
Nicely done, thanks